cargo/private/cargo_utils.bzl

```diff
@@ -151,18 +151,41 @@ def _resolve_repository_template(
 
     return template
 
-def get_rust_tools(cargo_template, rustc_template, host_triple, version):
+def cargo_home_path(repository_ctx):
+    """Define a path within the repository to use in place of `CARGO_HOME`
+
+    Args:
+        repository_ctx (repository_ctx): The rules context object
+
+    Returns:
+        path: The path to a directory to use as `CARGO_HOME`
+    """
+    return repository_ctx.path(".cargo_home")
+
+def get_rust_tools(cargo_template, rustc_template, host_triple, version, repository_ctx):
     """Retrieve `cargo` and `rustc` labels based on the host triple.
 
     Args:
         cargo_template (str): A template used to identify the label of the host `cargo` binary.
         rustc_template (str): A template used to identify the label of the host `rustc` binary.
         host_triple (struct): The host's triple. See `@rules_rust//rust/platform:triple.bzl`.
         version (str): The version of Cargo+Rustc to use.
+        repository_ctx (repository_ctx): The rule's context object.
 
     Returns:
         struct: A struct containing the labels of expected tools
     """
+
+    # This is a bit hidden but to ensure Cargo behaves consistently based
+    # on the user provided config file, the config file is installed into
+    # the `CARGO_HOME` path. This is done so here since fetching tools
+    # is expected to always occur before any subcommands are run.
+    if repository_ctx.attr.isolated and repository_ctx.attr.cargo_config:
```

Comment:
This functionality seems unrelated to the rest of `get_rust_tools`. Could this perhaps go in a separate function?

Reply:
The reason that I left this in is that it's used no where else and would be a function with side effects (creating the symlink). Do you have a suggestion for the name of such a function? I am happy to move it if you feel strongly about it.

Also, this function would take a `repository_ctx`. I so this would mean that would need to continue to be passed to the `get_rust_tools`. This is one of the places where there is fallout from not passing the `repository_ctx` that I mentioned above.